common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -60,30 +66,37 @@ public void recycle(Object object) {
         if (maxCapacity < 0) {
             maxCapacity = DEFAULT_INITIAL_MAX_CAPACITY;
         }
-
         DEFAULT_MAX_CAPACITY = maxCapacity;
 
+        MAX_SHARED_CAPACITY_FACTOR = max(DEFAULT_MAX_SHARED_CAPACITY_FACTOR,
```

Comment:
consider just using 2 here to be consistent with the limit imposed below for `LINK_CAPACITY`